Dockerfile.cpu

```diff
@@ -2,37 +2,49 @@
 
 FROM ubuntu:22.04 AS cpu-test-1
 
-RUN apt-get update -y \
-    && apt-get install -y curl git wget vim numactl gcc-12 g++-12 python3 python3-pip libtcmalloc-minimal4 libnuma-dev \
+RUN --mount=type=cache,target=/var/cache/apt \
+    apt-get update -y \
+    && apt-get install -y curl ccache git wget vim numactl gcc-12 g++-12 python3 python3-pip libtcmalloc-minimal4 libnuma-dev \
     && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-12 10 --slave /usr/bin/g++ g++ /usr/bin/g++-12
 
 # https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/performance_tuning/tuning_guide.html
 # intel-openmp provides additional performance improvement vs. openmp
 # tcmalloc provides better memory allocation efficiency, e.g, holding memory in caches to speed up access of commonly-used objects.
-RUN pip install intel-openmp
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install intel-openmp
 
 ENV LD_PRELOAD="/usr/lib/x86_64-linux-gnu/libtcmalloc_minimal.so.4:/usr/local/lib/libiomp5.so:$LD_PRELOAD"
 
 RUN echo 'ulimit -c 0' >> ~/.bashrc
 
 RUN pip install https://intel-extension-for-pytorch.s3.amazonaws.com/ipex_dev/cpu/intel_extension_for_pytorch-2.4.0%2Bgitfbaa4bc-cp310-cp310-linux_x86_64.whl
 
-RUN pip install --upgrade pip \
-    && pip install wheel packaging ninja "setuptools>=49.4.0" numpy
+ENV PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=bind,src=requirements-build.txt,target=requirements-build.txt \
+    pip install --upgrade pip && \
+    pip install -r requirements-build.txt
```

Comment:
Overall LGTM, thanks for the improvement! 

The only thing I concerned is using ```requirements-build.txt``` will install torch with many CUDA dependencies, which looks a bit redundant and increases download time.

Reply:
It will not: `PIP_EXTRA_INDEX_URL` is set before the RUN command, resulting in the following dependencies being installed:

```
Package           Version
----------------- ---------
cmake             3.30.2
filelock          3.15.4
fsspec            2024.6.1
Jinja2            3.1.4
MarkupSafe        2.1.5
mpmath            1.3.0
networkx          3.3
ninja             1.11.1.1
packaging         24.1
pip               24.1.2
setuptools        70.3.0
sympy             1.13.1
torch             2.4.0+cpu
typing_extensions 4.12.2
wheel             0.43.0
```

One possible issue might come up with some package managers such as `uv`:

```bash
export UV_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu
uv pip install -r requirements-build.txt
```

will fail due to [the way `uv` resolves local version identifiers](https://github.com/astral-sh/uv/blob/main/PIP_COMPATIBILITY.md#local-version-identifiers): in order for this to succeed, the `+cpu` specifier should be explicitly added (`torch==2.40+cpu`).

Reply:
Note that this `PIP_EXTRA_INDEX` url also works for PEP517 style builds, so running

`PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu pip install git+https://github.com/vllm-project/vllm` will also install the cpu version when building torch.